test/fbamodel_importer_test.py

```diff
@@ -0,0 +1,174 @@
+            print('Test workspace was deleted')
+
+    @classmethod
+    def delete_shock_node(cls, node_id):
+        header = {'Authorization': 'Oauth {0}'.format(cls.token)}
+        requests.delete(cls.shockURL + '/node/' + node_id, headers=header,
+                        allow_redirects=True)
+        print('Deleted shock node ' + node_id)
+
+    def getWsClient(self):
+        return self.__class__.wsClient
+
+    def getWsName(self):
+        if hasattr(self.__class__, 'wsName'):
+            return self.__class__.wsName
+        suffix = int(time.time() * 1000)
+        wsName = "test_kb_uploadmethods_" + str(suffix)
+        ret = self.getWsClient().create_workspace({'workspace': wsName})  # noqa
+        self.__class__.wsName = wsName
+        return wsName
+
+    def getImpl(self):
+        return self.__class__.serviceImpl
+
+    def getContext(self):
+        return self.__class__.ctx
+
+    def mock_download_staging_file(params):
+        print 'Mocking DataFileUtilClient.download_staging_file'
+        print params
+
+        fq_filename = params.get('staging_file_subdir_path')
+        fq_path = os.path.join('/kb/module/work/tmp', fq_filename)
+        shutil.copy(os.path.join("data", fq_filename), fq_path)
+
+        return {'copy_file_path': fq_path}
+
+    @patch.object(DataFileUtil, "download_staging_file", side_effect=mock_download_staging_file)
+    def test_bad_as_media_from_staging(self):
+        invalid_params = {
+            'file_type': 'sbml',
+            'workspace_name': self.getWsName(),
+            'fba_model_name': 'MyModel'
+        }
+        with self.assertRaisesRegexp(
+                    ValueError, 'Required parameter "model_file" is missing'):
+            self.getImpl().import_file_as_fba_model_from_staging(
+                self.getContext(), invalid_params)
+
+        invalid_params = {
+            'model_file': 'test_model-reactions.tsv',
+            'file_type': 'tsv',
+            'workspace_name': self.getWsName(),
+            'fba_model_name': 'MyModel'
+        }
+        with self.assertRaisesRegexp(
+                ValueError, 'A compound file is required for tsv upload.'):
+            self.getImpl().import_file_as_fba_model_from_staging(
+                self.getContext(), invalid_params)
+
+        invalid_params = {
+            'model_file': 'test_model-reactions.tsv',
+            'file_type': 'csv',
+            'workspace_name': self.getWsName(),
+            'fba_model_name': 'MyModel'
+        }
+        with self.assertRaisesRegexp(
+                ValueError, '"csv" is not a valid import file_type'):
+            self.getImpl().import_file_as_fba_model_from_staging(
+                self.getContext(), invalid_params)
+
+    @patch.object(DataFileUtil, "download_staging_file", side_effect=mock_download_staging_file)
+    def test_import_as_media_from_staging(self, download_staging_file):
+
+        params = {
```

Comment:
I would add a test with genome and biomass just making sure those params actually works.